src/routes/Dashboard/TaskViewEditModal.js

```diff
@@ -0,0 +1,308 @@
+import * as React from "react";
+
+import api from "../../api/axios";
+
+import Spinner from "../../components/Spinner";
+
+import * as Mui from "@mui/material";
+
+import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
+import { CircleOutlined } from "@mui/icons-material";
+
+import { ReactComponent as InboxIcon } from "../../icons/TaskEditModal/inbox.svg";
+import { ReactComponent as UserIcon } from "../../icons/TaskEditModal/user.svg";
+import { ReactComponent as DateIcon } from "../../icons/TaskEditModal/date.svg";
+import { ReactComponent as FlagIcon } from "../../icons/TaskEditModal/flag.svg";
+import { ReactComponent as PlusIcon } from "../../icons/TaskEditModal/plus.svg";
+import { ReactComponent as MoveUpIcon } from "../../icons/TaskEditModal/move_up.svg";
+import { ReactComponent as MoveDownIcon } from "../../icons/TaskEditModal/move_down.svg";
+import { ReactComponent as MenuIcon } from "../../icons/TaskEditModal/menu.svg";
+import { ReactComponent as CloseIcon } from "../../icons/TaskEditModal/close.svg";
+
+import { ReactComponent as PriorityIcon } from "../../icons/icon_priority.svg";
+
+export default function TaskViewEditModal({
+  open,
+  closeTaskModal,
```

Comment:
I suggest we rename this to 'onClose' as this is an event that should be handled by parent component.